File: 最新Scanner SDK 20240305/DemoScanner20240305/ScannerDemo/app/src/main/java/com/example/scanmanagerdemo/GetBitmapActivity.java
package com.example.scanmanagerdemo;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class GetBitmapActivity extends AppCompatActivity {

    private String action_scanner_capture_image_result = "scanner_capture_image_result";

    private ImageView imageView ;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_get_bitmap);
        imageView = findViewById(R.id.iv_bitmap);



        findViewById(R.id.btn_get_bitmap).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getBitmap();
            }
        });
        
//        //开启或关闭 [应用标识符 ( GS1 、DataMatrix Code128 )]
//        int [] keyInt = new int[]{PropertyID.LABEL_SEPARATOR_ENABLE};
//        int [] valueInt = new int[]{1};//0：关闭   1：开启
//        new ScanManager().setPropertyInts(keyInt,valueInt);

    }

    private void getBitmap(){
        //发送广播---取图
        String action = "action.scanner_capture_image" ;
        Intent intentImage = new Intent(action);
        sendBroadcast(intentImage);
    }

    @Override
    protected void onResume() {
        super.onResume();
        setReceiverRegister();
    }


    @Override
    protected void onPause() {
        super.onPause();
        unregisterReceiver(mGetBitmapReceiver);//注销广播
    }


    /**
     *广播接收器
     */
    private BroadcastReceiver mGetBitmapReceiver = new BroadcastReceiver() {

        @Override
        public void onReceive(Context context, Intent intent) {
            String action  = intent.getAction();
            if(action.equals(action_scanner_capture_image_result)) {
                byte[] bmp = intent.getByteArrayExtra("bitmapBytes");
                if(bmp != null && bmp.length > 1) {

                    Bitmap mBitmap = BitmapFactory.decodeByteArray(bmp, 0, bmp.length).copy(Bitmap.Config.RGB_565, true);
                    imageView.setImageBitmap(mBitmap);

                }else {
                    Toast.makeText(GetBitmapActivity.this, "获取图片失败 " , Toast.LENGTH_SHORT).show();
                }
            }
        }
    };

    /**
     *   注册广播
     */
    private void setReceiverRegister() {
        try {
            IntentFilter filter = new IntentFilter();
            filter.addAction(action_scanner_capture_image_result);
            registerReceiver(mGetBitmapReceiver, filter);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

}

File: runnable_demo/app/src/main/java/com/rtlink/runnable_demo/GetBitmapActivity.kt
package com.rtlink.runnable_demo

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class GetBitmapActivity : AppCompatActivity() {
    private val action_scanner_capture_image_result = "scanner_capture_image_result"

    private var imageView: ImageView? = null
    private lateinit var btn: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_get_bitmap)
        imageView = findViewById(R.id.iv_bitmap)
        btn = findViewById(R.id.btn_get_bitmap)

        btn.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                System.out.println(" ------------------------------------ ")
//                this.bitmap
            }
        })

        //        //开启或关闭 [应用标识符 ( GS1 、DataMatrix Code128 )]
//        int [] keyInt = new int[]{PropertyID.LABEL_SEPARATOR_ENABLE};
//        int [] valueInt = new int[]{1};//0：关闭   1：开启
//        new ScanManager().setPropertyInts(keyInt,valueInt);
    }

    private val bitmap: Unit
        get() {
            //发送广播---取图
            val action = "action.scanner_capture_image"
            val intentImage = Intent(action)
            sendBroadcast(intentImage)
        }

    override fun onResume() {
        super.onResume()
        setReceiverRegister()
    }


    override fun onPause() {
        super.onPause()
        unregisterReceiver(mGetBitmapReceiver) //注销广播
    }


    /**
     * 广播接收器
     */
    private val mGetBitmapReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            val action = intent.action
            if (action == action_scanner_capture_image_result) {
                val bmp = intent.getByteArrayExtra("bitmapBytes")
                if (bmp != null && bmp.size > 1) {
                    val mBitmap = BitmapFactory.decodeByteArray(bmp, 0, bmp.size)
                        .copy(Bitmap.Config.RGB_565, true)
                    imageView!!.setImageBitmap(mBitmap)
                } else {
                    Toast.makeText(this@GetBitmapActivity, "获取图片失败 ", Toast.LENGTH_SHORT)
                        .show()
                }
            }
        }
    }

    /**
     * 注册广播
     */
    private fun setReceiverRegister() {
        try {
            val filter = IntentFilter()
            filter.addAction(action_scanner_capture_image_result)
            registerReceiver(mGetBitmapReceiver, filter)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}